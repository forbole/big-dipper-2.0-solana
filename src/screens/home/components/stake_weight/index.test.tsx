import React from 'react';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import {
  StakeWeightDocument,
} from '@graphql/types';
import StakeWeight from '.';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);

const mockStakeWeight = jest.fn().mockResolvedValue({
  data: {
    validatorStatusAggregate: {
      aggregate: {
        sum: {
          activatedStake: 384573204910145540,
        },
      },
    },
    supplyInfo: [
      {
        total: 517013282309897200,
      },
    ],
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Home/StakeWeight', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      StakeWeightDocument,
      mockStakeWeight,
    );

    const component = renderer.create(
      <ApolloProvider client={mockClient}>
        <MockTheme>
          <StakeWeight />
        </MockTheme>
      </ApolloProvider>,
    );

    await wait();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
