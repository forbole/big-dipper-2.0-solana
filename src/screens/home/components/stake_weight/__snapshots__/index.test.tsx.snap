// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/StakeWeight matches snapshot 1`] = `
<div
  className="makeStyles-root makeStyles-root"
>
  <h2
    className="MuiTypography-root makeStyles-label MuiTypography-h2"
  >
    stakeWeight
  </h2>
  <div
    className="makeStyles-data"
  >
    <div
      className="data__item"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
        384,573,204,910.14 DARIC
      </h4>
      <div
        className="MuiTypography-root MuiTypography-caption"
      >
        weighted
      </div>
      <div
        className="MuiTypography-root MuiTypography-caption"
      >
        74%
      </div>
    </div>
    <div
      className="data__item"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
        1.32 DARIC
      </h4>
      <div
        className="MuiTypography-root MuiTypography-caption"
      >
        nonWeighted
      </div>
      <div
        className="MuiTypography-root MuiTypography-caption"
      >
        26%
      </div>
    </div>
  </div>
  <div
    className="makeStyles-content"
  >
    <div
      className="recharts-wrapper"
      style={
        Object {
          "cursor": "default",
          "height": 100,
          "position": "relative",
          "width": 200,
        }
      }
    >
      <svg
        className="recharts-surface"
        cx="50%"
        cy={100}
        height={100}
        version="1.1"
        viewBox="0 0 200 100"
        width={200}
      >
        <defs>
          <clipPath
            id="recharts1-clip"
          >
            <rect
              height={90}
              width={190}
              x={5}
              y={5}
            />
          </clipPath>
        </defs>
        <g
          className="recharts-layer recharts-pie"
        >
          <g
            className="recharts-layer"
          >
            <g
              className="recharts-layer recharts-pie-sector"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            />
            <g
              className="recharts-layer recharts-pie-sector"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            />
          </g>
        </g>
      </svg>
    </div>
    <div
      className="makeStyles-legends"
    >
      <div
        className="legends__item"
      >
        <span
          className="MuiTypography-root MuiTypography-caption"
        >
          weighted
        </span>
      </div>
      <div
        className="legends__item"
      >
        <span
          className="MuiTypography-root MuiTypography-caption"
        >
          nonWeighted
        </span>
      </div>
    </div>
  </div>
</div>
`;
